example/c/include/ICU4XDataProvider_type.h

```diff
@@ -0,0 +1,17 @@
+#ifndef ICU4XDataProvider_type_H
```

Comment:
Can you explain why this is superior to putting the forward declarations in the regular header files? I don't think I've seen this pattern very much of having a header file that doesn't have any real declarations in it.

Reply:
My original plan for HIR was that the HIR migration should produce no changes to generated code, however I think for the C backend I don't actually care that much.

I do think a nice property of this is that it forces users to import all the headers they need, which is in the long run less brittle.

Reply:
In most cases you're just replacing a pre-existing forward-declaration with the new import of the `_type` file.

Reply:
I guess what I'm saying is, it's fine with me to remove the implicit includes, but it seems like in-file forward declarations are the more common style. Are there examples you've seen of this sort of `_type.h` files in C/C++?

Reply:
I haven't; I've seen this pattern in C++ header only situations so I was copying that.

(like I said I think we will need to follow this pattern in C++ unfortunately)